x-pack/plugins/index_lifecycle_management/public/application/services/api.ts

```diff
@@ -37,9 +34,14 @@ export const useLoadNodeDetails = (selectedNodeAttrs: string) => {
   });
 };
 
-export async function loadIndexTemplates() {
-  return await sendGet(`templates`);
-}
+export const useLoadIndexTemplates = (legacy: boolean = false) => {
+  return useRequest<Array<{ name: string; settings: IndexSettings }>>({
+    path: 'templates',
+    query: { legacy },
+    method: 'get',
+    initialData: [],
```

Comment:
I see that we have this `initialData` in multiple places in the file. It does look strange to me because this "fakes" the use request expected behavior.
We want initially to have the data set to `null` and only when the request fulfills the data changes to whatever comes back from the server.
Providing an `initialData` looks like a hack that could be a source of bugs.

I think I overlooked that prop when reviewing the `useRequest` hook update and I would be in favor of removing it but maybe I am missing something? cc @cjcenizal

Reply:
@sebelga I don't recall the origin story of `initialData` but looking at the code again I'm inclined to agree. I'm not sure why we'd need this. I opened https://github.com/elastic/kibana/issues/78367 to track its removal.